examples/textbooks/generate_fineweb.py

```diff
@@ -0,0 +1,249 @@
+import asyncio
+import json
+import os
+import time
+from dataclasses import asdict, dataclass
+
+import pandas as pd
+from datasets import load_dataset
+from huggingface_hub import AsyncInferenceClient
+from tqdm.asyncio import tqdm_asyncio
+from transformers import AutoTokenizer, HfArgumentParser
+
+import wandb
+from llm_swarm import LLMSwarm, LLMSwarmConfig
+
+HF_TOKEN = os.environ.get("HF_TOKEN", None)
+
+
+@dataclass
+class Args:
+    # gneration parameters
+    tgi_instances: int = 10
+    """Number of TGI instances to use"""
+    model_name: str = "mistralai/Mixtral-8x7B-Instruct-v0.1"
+    """Dataset containing the prompts"""
+    max_new_tokens: int = 2500
+    """Max new tokens"""
+    temperature: float = 0.6
+    """Generation temperature"""
+    top_p: float = 0.95
+    """Generation top_p"""
+    top_k: int = 50
+    """Generation top_k"""
+    repetition_penalty: float = 1.2
+    """Generation repetition_penalty"""
+    # prompts dataset parameters
+    prompts_dataset: str = "HuggingFaceTB/fw_prompts_data_textbook"
+    """Model to prompt"""
+    max_samples: int = 5000
+    """The maximum umber of samples to generate (use -1 for all))"""
+    start_sample: int = -1
+    """First sample to process"""
+    end_sample: int = -1
+    """Last sample to process"""
+    num_proc: int = 36
+    """Number of processes for loading the dataset"""
+    prompt_column: str = "prompt_blogpost"
+    """Name of the column containing the prompt (from 'prompt_wikihow', 'prompt_textbook_narrative', 'prompt_textbook_academic', 'prompt_blogpost'"""
+    shuffle_dataset: bool = False
+    """Whether to shuffle the prompts"""
+    # logging parameters
+    repo_id: str = "HuggingFaceTB/fw_generations_test"
+    """The repo id to push to"""
+    checkpoint_path: str = "./fw_data"
+    """Path for saving intermediate generations"""
+    checkpoint_interval: int = 10_000
+    """Interval for saving intermediate generations"""
+    wandb_username: str = "loubnabnl"
+    """Wandb username"""
+    push_to_hub: bool = True
+    """Whether to push to hub"""
+
+
+parser = HfArgumentParser((Args, LLMSwarmConfig))
+args, isc = parser.parse_args_into_dataclasses()
+print(args)
+
+# overwrite model and number of instances
+isc.model = args.model_name
+isc.instances = args.tgi_instances
+tokenizer = AutoTokenizer.from_pretrained(args.model_name)
+SEED = 42
+
+ds = load_dataset(
+    args.prompts_dataset, token=HF_TOKEN, split="train", num_proc=args.num_proc
```

Comment:
What I typically do is add a `debug: bool = False` in the Args, and 

```num_proc=1 if args.debug else multiprocessing.cpu_count()```. This gives some flexibility in debugging and allows using the whole CPUs when running at scale.

Reply:
Yeah good point, I changed it